templates/paypal-payments-upgrade-notice.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * Show PayPal Payments upgrade notice on plugins page
+ *
+ * @package woocommerce-paypal-express-checkout/templates
+ */
+
+$paypal_payments_path         = 'woocommerce-paypal-payments/woocommerce-paypal-payments.php';
+$is_installed_paypal_payments = array_key_exists( $paypal_payments_path, get_plugins() );
+$is_active_paypal_payments    = is_plugin_active( $paypal_payments_path );
+
+// Generate Install / Activation / Config links.
+$paypal_payments_install_link = wp_nonce_url(
+	add_query_arg(
+		array(
+			'action' => 'install-plugin',
+			'plugin' => dirname( $paypal_payments_path ),
+		),
+		admin_url( 'update.php' )
+	),
+	'install-plugin_' . dirname( $paypal_payments_path )
+);
+
+$paypal_payments_activate_link = wp_nonce_url(
+	add_query_arg(
+		array(
+			'action' => 'activate',
+			'plugin' => $paypal_payments_path,
+		),
+		admin_url( 'plugins.php' )
+	),
+	'activate-plugin_' . $paypal_payments_path
+);
+
+// Template for rendering buttons.
+$button_data = array(
+	'id'         => null,
+	'href'       => null,
+	'text'       => null,
+	'attributes' => array(),
+);
+
+if ( $is_active_paypal_payments && $is_active_paypal_payments ) {
+
+	// PayPal Payments installed and active.
+	$button_data['id']   = 'ppec-configure-paypal-payments';
+	$button_data['href'] = admin_url( 'admin.php?page=wc-settings&tab=checkout&section=ppcp-gateway' );
+	$button_data['text'] = 'Configure PayPal Payments now';
+
+} elseif ( $is_installed_paypal_payments ) {
+
+	// PayPal Payments installed, but inactive.
+	$button_data['id']         = 'ppec-activate-paypal-payments';
+	$button_data['href']       = $paypal_payments_activate_link;
+	$button_data['text']       = 'Activate PayPal Payments now';
+	$button_data['attributes'] = array(
```

Comment:
I was thinking about this part specifically, and while every button only cares about its own id/href/text, this button in particular needs some context (which, in turn, is the id/href/text for another one). Because of that, it might be less complicated if we just print *both* buttons ("Activate" and "Upgrade") in the HTML and use the JS to hide/show the buttons accordingly.

What do you think? This could also slightly reduce some of the logic in the JS and the template itself, and also some of the "duplication" of the `$button_data` here for the `$is_installed_paypal_payments = true` case.

Reply:
We can go with your suggestion to add all buttons and hide specific ones via JS. However, we would need to add HTML for for 4 buttons. But yes, it would reduce the logic added on the template.

1. Install
2. Activate
3. Configure
4. Learn More

If you think this is better, I'm happy to make the changes. I really don't have a preference in this case.

Reply:
"Learn More" is already in the HTML, so that wouldn't be too different. As for the others, we don't need to actually _print_ all of those 3 all the time (Install and Activate, for example).

If that's ok with you, I do like the idea of printing the HTML 👍

Reply:
Changes updated